packages/client/tests/functional/_utils/setupTestSuiteEnv.ts

```diff
@@ -115,24 +116,76 @@ export async function setupTestSuiteDatabase(
+
+      // Note: default hostname is `http://127.0.0.1:8787`
+      // The custom port is set in packages/client/tests/functional/_utils/wrangler.toml
+      d1Client = connectD1('MY_DATABASE', { hostname: 'http://127.0.0.1:9090' })
+
+      // Delete existing tables
+      const existingItems = (await d1Client.prepare(`PRAGMA main.table_list;`).run()).results
+      for (const item of existingItems) {
+        const batch: D1PreparedStatement[] = []
+        if (item.name === '_cf_KV' || item.name === 'sqlite_schema') {
+          continue
+        } else if (item.name === 'sqlite_sequence') {
+          batch.push(d1Client.prepare('DELETE FROM `sqlite_sequence`;'))
+        } else if (item.type === 'view') {
+          batch.push(d1Client.prepare(`DROP VIEW "${item.name}";`))
+        } else {
+          batch.push(d1Client.prepare(`DROP TABLE "${item.name}";`))
+        }
+
+        await d1Client.batch(batch)
+      }
 
-    // we reuse and clean the db when it is explicitly required
-    if (process.env.TEST_REUSE_DATABASE === 'true') {
-      dbPushParams.push('--force-reset')
-    }
+      // Use `migrate diff` to get the DDL statements
+      const diffResult = await execa(
+        '../cli/src/bin.ts',
+        ['migrate', 'diff', '--from-empty', '--to-schema-datamodel', schemaPath, '--script'],
+        {
+          env: {
+            DEBUG: process.env.DEBUG,
+          },
+        },
+      )
+      const sqlStatements = diffResult.stdout
+
+      // Execute the DDL statements
+      for (const sqlStatement of sqlStatements.split(';')) {
+        console.log({ sqlStatementsqlStatement: sqlStatement })
+        if (sqlStatement.includes('CREATE ')) {
+          await d1Client.prepare(sqlStatement).run()
+        } else if (sqlStatement === '\n') {
+          // Ignore
+        } else {
+          console.debug(`Skipping ${sqlStatement} as it is not a CREATE statement`)
+        }
+      }
+    } else {
+      const dbPushParams = ['--schema', schemaPath, '--skip-generate']
 
-    await DbPush.new().parse(dbPushParams)
+      // we reuse and clean the db when it is explicitly required
+      if (process.env.TEST_REUSE_DATABASE === 'true') {
+        dbPushParams.push('--force-reset')
+      }
 
-    if (
-      suiteConfig.matrixOptions.driverAdapter === AdapterProviders.VITESS_8 ||
-      suiteConfig.matrixOptions.driverAdapter === AdapterProviders.JS_PLANETSCALE
-    ) {
-      // wait for vitess to catch up, corresponds to TABLET_REFRESH_INTERVAL in docker-compose.yml
-      await new Promise((r) => setTimeout(r, 1_000))
+      await DbPush.new().parse(dbPushParams)
+
+      if (
+        suiteConfig.matrixOptions.driverAdapter === AdapterProviders.VITESS_8 ||
+        suiteConfig.matrixOptions.driverAdapter === AdapterProviders.JS_PLANETSCALE
+      ) {
+        // wait for vitess to catch up, corresponds to TABLET_REFRESH_INTERVAL in docker-compose.yml
+        await new Promise((r) => setTimeout(r, 1_000))
+      }
```

Comment:
Could we move this on the block below as two functions into a separate file, for example `setupTestSuiteEnvD1.ts` or something similar?

Reply:
Yes, I think it can be done later

Reply:
I added `setupTestSuiteDatabaseD1` for now